pandas/core/internals/concat.py

```diff
@@ -232,6 +227,28 @@ def dtype(self):
             return blk.dtype
         return ensure_dtype_can_hold_na(blk.dtype)
 
+    def is_valid_na_for(self, dtype: DtypeObj) -> bool:
+        """
+        Check that we are all-NA of a type/dtype that is compatible with this dtype.
```

Comment:
```suggestion
        Check that we are all-NA of a type/dtype that is compatible with this dtype.
        Augments `self.is_na` with an additional check of the type of NA values.
```

That's correct? 

(I was missing before that this basically is `self.is_na` but a bit stricter for a few cases)

Reply:
yes